docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
 # Developer Guide
 
 ## Design & implementation
+<!-- @@author iamchenjiajun -->
```

Comment:
You guys may consider add a content table with hyperlink and an introduction for your product and this document (developer guide), so that the readers can have a rough idea of your whole project and can choose to look into detail in specific sessions, instead of being thrown with architecture content/diagram right at the beginning.